src/utils/functions/getDonationLateStatus.ts

```diff
@@ -0,0 +1,49 @@
+import { DateTime } from 'luxon';
+
+enum ContactLateStatusEnum {
+  OnTime,
+  LateLessThirty,
+  LateMoreThirty,
+  LateMoreSixty,
+}
+
+function selectLaterDate(lateAt: string, pledgeStartDate: string): string {
+  const lateDate = DateTime.fromISO(lateAt);
+  const pledgeDate = DateTime.fromISO(pledgeStartDate);
+  return lateDate > pledgeDate ? lateAt : pledgeStartDate;
+}
+
+function determineStatusFromDays(daysDiff: number): ContactLateStatusEnum {
+  if (daysDiff < 0) {
+    return ContactLateStatusEnum.OnTime;
+  }
+  if (daysDiff < 30) {
+    return ContactLateStatusEnum.LateLessThirty;
+  }
+  if (daysDiff < 60) {
+    return ContactLateStatusEnum.LateMoreThirty;
+  }
+  return ContactLateStatusEnum.LateMoreSixty;
+}
+
+export function getDonationLateStatus(
+  lateAt?: string | null,
+  pledgeStartDate?: string | null,
+): ContactLateStatusEnum | undefined {
+  // Determine which date to use
+  const dateToUse =
```

Comment:
Can we give this a slightly more descriptive name? What do you think about `laterDate`?

```suggestion
  const laterDate =
```